lcd/config_downloader.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+import uuid
+import requests
+
+class ConfigDownloader:
+    def __init__(self, config_path, remote_url):
+        self.config_path = config_path
+        self.remote_url  = remote_url
+        self.api_key      = uuid.uuid4()
+
+    def run(self):
+        print "Setting up a new account"
+        print "please make sure an account with API key %s is configured" % self.api_key
+        ready = raw_input("press return when ready... ")
+        if ready != "":
+            print "aborting..."
+            exit()
+        self.download()
+        if self.response.status_code == requests.codes.ok:
+            print "writing config file to: " + self.config_path
+            self.save()
+            print "done."
+        else:
+            print "dowload failed with status: %s" % self.response.status_code
+
+    def download(self):
+        self.response = requests.get(self.remote_url, headers={'X-Api-Key': self.api_key})
```

Comment:
connection failures could also be one that's common place.
Meaning full error message is probably enough